game-app/game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbTokenGenerator.java

```diff
@@ -72,7 +73,8 @@ public TokenInfo generateToken(
    * @param userId The userId that the token was issued for.
    */
   public void revokeToken(final String token, final int userId) {
-    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement()
```

Comment:
What does 'disableDefaultUserClient' do, should we alwasy use that?
I suspect as well we may want to move this code into a common lib base-layer to make it easier/more generic.

Reply:
This seems to be the "fix" for some reason.
During debugging I was reproducing the exact same steps as the code using curl and my webbrowser. Using curl and the browser everything worked as expected only java was having some weird issues.
The only possible difference could be the headers the client sends, so I tried to disable some headers, and after removing the user agent header (which tells the HTTP server which browser is requesting the site) everything worked fine again.

Reply:
> I suspect as well we may want to move this code into a common lib base-layer to make it easier/more generic.

That would be an option of course but I don't think it's worth to invest too much time into this particular code.
Http Components 4 has started to become legacy code (HttpComponents 5 is out and calls the current code scheme the "classic" scheme, see https://hc.apache.org/httpcomponents-client-5.0.x/index.html ). We could consider migrating to the modern async approaches to this problem, or we could convert this code into a feign pattern, or migrate to the "new" HttpClient from the standard library which seems to have a nice API: https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html

It avoids having to wrap the code in try-with-resource statements and provides a nice builder API, so the new code would looks more like this:
```java
HttpClient client = HttpClients.newBuilder()
  .followRedirects(Redirect.NORMAL)
  .proxy(ProxySelector.of(new InetSocketAddress(ClientSetting.proxyHost.getValueOrThrow(), ClientSetting.proxyPort.getValueOrThrow())))
  .build();

 HttpRequest request = HttpRequest.newBuilder()
    .uri(URI.create("https://foo.com/"))
    .GET()
    .build();
  HttpResponse<String> response = client.send(request, BodyHandlers.ofString());
  System.out.println(response.statusCode());
  System.out.println(response.body());  
```
And instead of `BodyHandlers.ofString()` we could plug in our own yaml/json parser.

This way the API becomes slightly nicer and we could dump a dependency

Reply:
Thanks for the feedback, interesting to see there is already a new API.

IMO, best to keep the tech stack as consistent as we can and use Feign. Feign does not support 'put' though, therefore for file uploads we'll likely use httpcomponents.